File: backend/Models/Reservation.cs
namespace backend.Models
{
    public class Reservation
    {
        public int ReservationId { get; set; }
        public TimeOnly ReservationStart { get; set; }
        public TimeOnly ReservationEnd { get; set; }
        public DateOnly ReservationDate { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public DateTime? ReservedAt { get; set; }
        public double Price { get; set; }

        public int ObjectId { get; set; }
        public ObjectType ObjectType { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
    }
}

File: frontend/src/shared/types/models/reservation.ts
import { ObjectTypeDto } from "./objectType";

export interface ReservationDto {
    reservationId: number,
    reservationStart: string;
    reservationEnd: string;
    reservationDate: string;
    paymentStatus: string,
    reservedAt: string,
    price: number;
    objectType: ObjectTypeDto;
}

export interface CreateReservationData {
    reservationStart: string | null;
    reservationEnd: string | null;
    reservationDate: string;
    price: number | undefined;
    objectId: number | null;
    userId: number | null;
    isPaid: boolean;
}

export interface ReservationListsResponse {
    reservationsStart: string
    reservationsEnd: string
    freeHours: string[]
    reservedHours: string[]
    date: string
    isTournament: boolean
};